src/services/DexService.ts

```diff
@@ -141,98 +126,145 @@ export default class DexService {
     addPropertyToClass(
       baseStatsData,
       'hp',
       basePokemonData.baseStats.hp,
-      requestedFields as GraphQLSet<keyof StatsEntry>,
+      baseStatsRequestedFields,
       `${recursingAs ? `${recursingAs}.` : ''}baseStats.hp`
     );
     addPropertyToClass(
       baseStatsData,
       'attack',
       basePokemonData.baseStats.atk,
-      requestedFields as GraphQLSet<keyof StatsEntry>,
+      baseStatsRequestedFields,
       `${recursingAs ? `${recursingAs}.` : ''}baseStats.attack`
     );
     addPropertyToClass(
       baseStatsData,
       'defense',
       basePokemonData.baseStats.def,
-      requestedFields as GraphQLSet<keyof StatsEntry>,
+      baseStatsRequestedFields,
       `${recursingAs ? `${recursingAs}.` : ''}baseStats.defense`
     );
     addPropertyToClass(
       baseStatsData,
       'specialattack',
       basePokemonData.baseStats.spa,
-      requestedFields as GraphQLSet<keyof StatsEntry>,
+      baseStatsRequestedFields,
       `${recursingAs ? `${recursingAs}.` : ''}baseStats.specialattack`
     );
     addPropertyToClass(
       baseStatsData,
       'specialdefense',
       basePokemonData.baseStats.spd,
-      requestedFields as GraphQLSet<keyof StatsEntry>,
+      baseStatsRequestedFields,
       `${recursingAs ? `${recursingAs}.` : ''}baseStats.specialdefense`
     );
     addPropertyToClass(
       baseStatsData,
       'speed',
       basePokemonData.baseStats.spe,
-      requestedFields as GraphQLSet<keyof StatsEntry>,
+      baseStatsRequestedFields,
       `${recursingAs ? `${recursingAs}.` : ''}baseStats.speed`
     );
 
+    const evYieldsRequestedFields = Util.cast<GraphQLSet<keyof EvYieldsEntry>>(requestedFields);
+    addPropertyToClass(
+      evYieldsData,
+      'hp',
+      basePokemonData.evYields?.hp ?? 0,
+      evYieldsRequestedFields,
+      `${recursingAs ? `${recursingAs}.` : ''}evYields.hp`
+    );
+    addPropertyToClass(
+      evYieldsData,
+      'attack',
+      basePokemonData.evYields?.atk ?? 0,
+      evYieldsRequestedFields,
+      `${recursingAs ? `${recursingAs}.` : ''}evYields.attack`
+    );
+    addPropertyToClass(
+      evYieldsData,
+      'defense',
+      basePokemonData.evYields?.def ?? 0,
+      evYieldsRequestedFields,
+      `${recursingAs ? `${recursingAs}.` : ''}evYields.defense`
+    );
+    addPropertyToClass(
+      evYieldsData,
+      'specialattack',
+      basePokemonData.evYields?.spa ?? 0,
+      evYieldsRequestedFields,
+      `${recursingAs ? `${recursingAs}.` : ''}evYields.specialattack`
+    );
+    addPropertyToClass(
+      evYieldsData,
+      'specialdefense',
+      basePokemonData.evYields?.spd ?? 0,
+      evYieldsRequestedFields,
+      `${recursingAs ? `${recursingAs}.` : ''}evYields.specialdefense`
+    );
+    addPropertyToClass(
+      evYieldsData,
+      'speed',
+      basePokemonData.evYields?.spe ?? 0,
+      evYieldsRequestedFields,
+      `${recursingAs ? `${recursingAs}.` : ''}evYields.speed`
+    );
```

Comment:
```suggestion
    addPropertyToClass(
      evYieldsData,
      'hp',
      basePokemonData.evYields.hp,
      evYieldsRequestedFields,
      `${recursingAs ? `${recursingAs}.` : ''}evYields.hp`
    );
    addPropertyToClass(
      evYieldsData,
      'attack',
      basePokemonData.evYields.atk,
      evYieldsRequestedFields,
      `${recursingAs ? `${recursingAs}.` : ''}evYields.attack`
    );
    addPropertyToClass(
      evYieldsData,
      'defense',
      basePokemonData.evYields.def,
      evYieldsRequestedFields,
      `${recursingAs ? `${recursingAs}.` : ''}evYields.defense`
    );
    addPropertyToClass(
      evYieldsData,
      'specialattack',
      basePokemonData.evYields.spa,
      evYieldsRequestedFields,
      `${recursingAs ? `${recursingAs}.` : ''}evYields.specialattack`
    );
    addPropertyToClass(
      evYieldsData,
      'specialdefense',
      basePokemonData.evYields.spd,
      evYieldsRequestedFields,
      `${recursingAs ? `${recursingAs}.` : ''}evYields.specialdefense`
    );
    addPropertyToClass(
      evYieldsData,
      'speed',
      basePokemonData.evYields.spe,
      evYieldsRequestedFields,
      `${recursingAs ? `${recursingAs}.` : ''}evYields.speed`
    );
```

Once `evYields` has been added everywhere and the `?` in the `pokemon.d.ts` has been removed this change needs to be applied as well